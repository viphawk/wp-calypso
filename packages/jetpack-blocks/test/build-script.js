/* eslint-disable import/no-nodejs-modules */
/**
 * External dependencies
 */
import { exec } from 'child_process';
import { join } from 'path';
import { readFile } from 'fs';

describe( 'jetpack-blocks-build', () => {
	test(
		'builds-as-expected',
		() =>
			new Promise( ( resolve, reject ) => {
				exec(
					`NODE_ENV='development' SOURCEMAP='none' node ${ join(
						__dirname,
						'..',
						'bin',
						'build.js'
					) } --test-build`,
					buildError => {
						if ( buildError ) {
							return reject( buildError );
						}
						readFile(
							join( __dirname, '..', 'dist', 'test.js' ),
							'utf8',
							( readError, contents ) => {
								if ( readError ) {
									return reject( readError );
								}
								expect( contents ).toMatchSnapshot();
								resolve();
							}
						);
					}
				);
			} ),
		30000
	);
} );
