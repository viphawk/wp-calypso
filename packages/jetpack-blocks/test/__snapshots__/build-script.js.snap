// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jetpack-blocks-build builds-as-expected 1`] = `
"(function(e, a) { for(var i in a) e[i] = a[i]; }(window, /******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./packages/jetpack-blocks/test/fixtures/test-build.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./node_modules/@babel/runtime/helpers/classCallCheck.js\\":
/*!***************************************************************!*\\\\
  !*** ./node_modules/@babel/runtime/helpers/classCallCheck.js ***!
  \\\\***************************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError(\\"Cannot call a class as a function\\");
  }
}

module.exports = _classCallCheck;

/***/ }),

/***/ \\"./node_modules/@babel/runtime/helpers/createClass.js\\":
/*!************************************************************!*\\\\
  !*** ./node_modules/@babel/runtime/helpers/createClass.js ***!
  \\\\************************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

function _defineProperties(target, props) {
  for (var i = 0; i < props.length; i++) {
    var descriptor = props[i];
    descriptor.enumerable = descriptor.enumerable || false;
    descriptor.configurable = true;
    if (\\"value\\" in descriptor) descriptor.writable = true;
    Object.defineProperty(target, descriptor.key, descriptor);
  }
}

function _createClass(Constructor, protoProps, staticProps) {
  if (protoProps) _defineProperties(Constructor.prototype, protoProps);
  if (staticProps) _defineProperties(Constructor, staticProps);
  return Constructor;
}

module.exports = _createClass;

/***/ }),

/***/ \\"./node_modules/@babel/runtime/helpers/defineProperty.js\\":
/*!***************************************************************!*\\\\
  !*** ./node_modules/@babel/runtime/helpers/defineProperty.js ***!
  \\\\***************************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

module.exports = _defineProperty;

/***/ }),

/***/ \\"./packages/jetpack-blocks/test/fixtures/test-build.js\\":
/*!*************************************************************!*\\\\
  !*** ./packages/jetpack-blocks/test/fixtures/test-build.js ***!
  \\\\*************************************************************/
/*! exports provided: destructuredAs, SomeClass, ohHeyJsx */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\"destructuredAs\\", function() { return destructuredAs; });
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\"SomeClass\\", function() { return SomeClass; });
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\"ohHeyJsx\\", function() { return ohHeyJsx; });
/* harmony import */ var _babel_runtime_helpers_classCallCheck__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! @babel/runtime/helpers/classCallCheck */ \\"./node_modules/@babel/runtime/helpers/classCallCheck.js\\");
/* harmony import */ var _babel_runtime_helpers_classCallCheck__WEBPACK_IMPORTED_MODULE_0___default = /*#__PURE__*/__webpack_require__.n(_babel_runtime_helpers_classCallCheck__WEBPACK_IMPORTED_MODULE_0__);
/* harmony import */ var _babel_runtime_helpers_createClass__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! @babel/runtime/helpers/createClass */ \\"./node_modules/@babel/runtime/helpers/createClass.js\\");
/* harmony import */ var _babel_runtime_helpers_createClass__WEBPACK_IMPORTED_MODULE_1___default = /*#__PURE__*/__webpack_require__.n(_babel_runtime_helpers_createClass__WEBPACK_IMPORTED_MODULE_1__);
/* harmony import */ var _babel_runtime_helpers_defineProperty__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(/*! @babel/runtime/helpers/defineProperty */ \\"./node_modules/@babel/runtime/helpers/defineProperty.js\\");
/* harmony import */ var _babel_runtime_helpers_defineProperty__WEBPACK_IMPORTED_MODULE_2___default = /*#__PURE__*/__webpack_require__.n(_babel_runtime_helpers_defineProperty__WEBPACK_IMPORTED_MODULE_2__);
/* harmony import */ var _wordpress_element__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(/*! @wordpress/element */ \\"@wordpress/element\\");
/* harmony import */ var _wordpress_element__WEBPACK_IMPORTED_MODULE_3___default = /*#__PURE__*/__webpack_require__.n(_wordpress_element__WEBPACK_IMPORTED_MODULE_3__);





var _destructured = _babel_runtime_helpers_defineProperty__WEBPACK_IMPORTED_MODULE_2___default()({}, 'destructured', 'destructured'),
    _destructured$destruc = _destructured.destructured,
    destructuredAs = _destructured$destruc === void 0 ? 'default' : _destructured$destruc;


var SomeClass =
/*#__PURE__*/
function () {
  function SomeClass() {
    _babel_runtime_helpers_classCallCheck__WEBPACK_IMPORTED_MODULE_0___default()(this, SomeClass);

    _babel_runtime_helpers_defineProperty__WEBPACK_IMPORTED_MODULE_2___default()(this, \\"instanceProp\\", null);

    _babel_runtime_helpers_defineProperty__WEBPACK_IMPORTED_MODULE_2___default()(this, \\"arrowMethod\\", function () {});
  }

  _babel_runtime_helpers_createClass__WEBPACK_IMPORTED_MODULE_1___default()(SomeClass, [{
    key: \\"method\\",
    value: function method() {}
  }]);

  return SomeClass;
}();

_babel_runtime_helpers_defineProperty__WEBPACK_IMPORTED_MODULE_2___default()(SomeClass, \\"thing\\", null);

var ohHeyJsx = Object(_wordpress_element__WEBPACK_IMPORTED_MODULE_3__[\\"createElement\\"])(\\"span\\", {
  role: \\"img\\",
  \\"aria-label\\": \\"Oh, hey!\\"
}, \\"\\\\uD83D\\\\uDC4B\\");

/***/ }),

/***/ \\"@wordpress/element\\":
/*!*****************************!*\\\\
  !*** external \\"wp.element\\" ***!
  \\\\*****************************/
/*! no static exports found */
/***/ (function(module, exports) {

module.exports = wp.element;

/***/ })

/******/ })));"
`;
